src/Umbraco.Web.UI.Client/src/common/services/navigation.service.js

```diff
@@ -288,9 +288,18 @@ function navigationService($rootScope, $routeParams, $log, $location, $q, $timeo
                 throw "args.tree cannot be null";
             }
 
-            if (mainTreeEventHandler) {
-                //returns a promise
-                return mainTreeEventHandler.syncTree(args);
+            if (mainTreeEventHandler) {                
+               
+                if (mainTreeEventHandler.syncTree) {
+                    //returns a promise,
+                    return mainTreeEventHandler.syncTree(args);
+                } else {
+                    //returns a promise, but wrap it in a time out,
+                    // on some occasions the syncTree method is not defined yet in the umb - tree directive(U4 - 11167)
+                    $timeout(function () {
```

Comment:
This is still not actually returning a promise :) this is executing a function after a timeout and that inner function is returning a value to the $timeout method, not to the caller of `syncTree`

I'm wondering if we can instead:

* Error check that `mainTreeEventHandler.syncTree` is defined and if not throw an exception - if we are expecting it to always be set
* eagerly ensure that `mainTreeEventHandler.syncTree` is set before `navigationService.syncTree` is ever executed

Reply:
Hi Shannon,

Maybe just suppress the error for now when syncTree is not defined. This bug has been present for a longer time. I could trace it back to 7.8.0, but I did not try earlier versions.

It also only happens when special conditions are met...eg. doing a hard refresh and with empty cache. And then not even every time.

Dave